### DISCUSSION TITLE
How to save completed workflow in oracle

### DISCUSSION THREAD

**Discussion by FounderHy** [argoproj/argo-workflows#7941] [Q&A]
In this config: https://argoproj.github.io/argo-workflows/workflow-controller-configmap.yaml, I just found database config of mysql and postgres, I want to know how to save workflow in oracle?

**Comment by terrytangyuan** [comment]
Check out the docs here: https://argoproj.github.io/argo-workflows/offloading-large-workflows/

**Reply by FounderHy** [reply]
> Check out the docs here: https://argoproj.github.io/argo-workflows/offloading-large-workflows/

Thankyou for reply, I have read the docs before, I know I can save workflow in mysql or postgres, I want to know can I save workflow in oracle database ?

**Reply by terrytangyuan** [reply]
I am not sure how compatible Oracle database is with Postgres or MySQL. The best way to find out is to give it a try.

**Reply by vitalyrychkov** [reply]
@terrytangyuan 

Hi, I have tried to cheat and put Oracle connection parameters instead of PostgreSQL, but the workflow controller fails at startup:
	time="2023-03-09T10:01:21.283Z" level=info msg="Persistence configuration enabled"
	time="2023-03-09T10:01:21.286Z" level=info msg="Get secrets 200"
	time="2023-03-09T10:01:21.289Z" level=info msg="Get secrets 200"
	time="2023-03-09T10:01:21.435Z" level=fatal msg="Failed to update config: EOF"
	
Normally the last line must be:
									level=info msg=Persistence Session created successfully

This log line is found in https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/config.go
but is not executed, because sqldb.CreateDBSession returns err:
	[...]		
	persistence := wfc.Config.Persistence
	if persistence != nil {
		log.Info("Persistence configuration enabled")
		session, tableName, err := sqldb.CreateDBSession(wfc.kubeclientset, wfc.namespace, persistence)
		if err != nil {
			return err
		}
		log.Info("Persistence Session created successfully")
	[...]		

The source https://github.com/argoproj/argo-workflows/blob/master/persist/sqldb/sqldb.go seems to be postgres or mysql specific, but I have no experience with Go to judge.

Some of our potential customers are using Oracle and have no PostgreSQL, would it be possible to add Oracle to the list of supported databases?
